addons/mail/static/tests/owl/component/notification_list_tests.js

```diff
@@ -0,0 +1,214 @@
+                            model: 'mail.channel',
+                            record_name: 'Channel 2020',
+                            res_id: 200,
+                        },
+                    }
+                ];
+            }
+            return this._super(...arguments);
+        }
+    });
+    await this.createNotificationListComponent({ filter: 'all' });
+    assert.containsN(document.body, '.o_ThreadPreview', 2,
+        "there should be two thread previews");
+    const threadPreviewElList = document.querySelectorAll('.o_ThreadPreview');
+    assert.strictEqual(
+        threadPreviewElList[0].querySelector(':scope .o_ThreadPreview_name').textContent,
+        'Channel 2020',
+        "First channel in the list should be the channel of 2020 (more recent)"
+    );
+    assert.strictEqual(
+        threadPreviewElList[1].querySelector(':scope .o_ThreadPreview_name').textContent,
+        'Channel 2019',
+        "First channel in the list should be the channel of 2019 (least recent)"
+    );
+});
+
+QUnit.test('reordering on new message received', async function (assert) {
+    assert.expect(4);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 100,
+                name: "Channel 2019",
+                message_unread_counter: 0,
+            }, {
+                channel_type: "channel",
+                id: 200,
+                name: "Channel 2020",
+                message_unread_counter: 0,
+            }],
+        },
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fetch_preview') {
+                return [
+                    {
+                        id: 100,
+                        last_message: {
+                            author_id: [100, `Author A`],
+                            body: `<p>Message A</p>`,
+                            channel_ids: [100],
+                            date: `2019-01-01 00:00:00`,
+                            id: 42,
+                            message_type: 'comment',
+                            model: 'mail.channel',
+                            record_name: 'Channel 2019',
+                            res_id: 100,
+                        },
+                    },
+                    {
+                        id: 200,
+                        last_message: {
+                            author_id: [200, `Author B`],
+                            body: `<p>Message B</p>`,
+                            channel_ids: [200],
+                            date: `2020-01-01 00:00:00`,
+                            id: 43,
+                            message_type: 'comment',
+                            model: 'mail.channel',
+                            record_name: 'Channel 2020',
+                            res_id: 200,
+                        },
+                    }
```

Comment:
I would strongly avoid mixing same id value for different objects with different model: it's quite confusing to have channel, message, and partner (author), all with same id 100 or 200.

I know this could happen in practice, but IMO it just makes this test harder to read